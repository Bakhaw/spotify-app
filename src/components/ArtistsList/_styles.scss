.ArtistsList {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  &__ListItem {
    margin: 30px 0;

    p {
      margin: 0;
      color: #fff;
      text-align: center;
    }
  }
}
