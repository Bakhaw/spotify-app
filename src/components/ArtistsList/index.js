import React from 'react';
import { Link } from 'react-router-dom';

import Avatar from '../Avatar';

function ArtistsList({ data }) {
  return (
    <ul className='ArtistsList'>
      {data.map((artist, i) => (
        <li key={i} className='ArtistsList__ListItem'>
          <Link to={`/artist/${artist.id}`}>
            <Avatar artist={artist} />
            <p>{artist.name}</p>
          </Link>
        </li>
      ))}
    </ul>
  );
}

export default ArtistsList;
