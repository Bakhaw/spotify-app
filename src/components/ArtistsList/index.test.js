import React from 'react'
import { BrowserRouter} from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ArtistsList from '.';

test('components/ArtistsList/', () => {
  const data = [
    {
      artist: {
        id: 1,
        name: 'Damso'
      }
    }
  ]

  const { container } = render(
    <BrowserRouter>
      <ArtistsList data={data} />
    </BrowserRouter>
  );
  expect(container.firstChild).toMatchSnapshot();
})