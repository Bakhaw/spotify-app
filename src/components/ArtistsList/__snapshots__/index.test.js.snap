// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ArtistsList/ 1`] = `
<ul
  class="ArtistsList"
>
  <li
    class="ArtistsList__ListItem"
  >
    <a
      href="/artist/undefined"
    >
      <div>
        <img
          class="Avatar"
        />
      </div>
      <p />
    </a>
  </li>
</ul>
`;
